cloud_sql/instance_psc/main.tf

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START cloud_sql_mysql_instance_psc]
+resource "google_sql_database_instance" "mysql_instance_psc" {
+  name             = "psc-enabled-mysql-instance"
+  region           = "us-central1"
+  database_version = "MYSQL_8_0"
+  settings {
+    tier              = "db-f1-micro"
+    availability_type = "REGIONAL"
+    backup_configuration {
+      enabled            = true
+      binary_log_enabled = true
+    }
+    ip_configuration {
+      psc_config {
+        psc_enabled               = true
+        allowed_consumer_projects = []
+      }
+      ipv4_enabled = false
+    }
+  }
+}
+# [END cloud_sql_mysql_instance_psc]
+
+# [START cloud_sql_postgres_instance_psc]
+resource "google_sql_database_instance" "postgres_instance_psc" {
```

Comment:
After splitting the file, rename each `google_sql_database_instance` to `default` per samples' best practice guidelines.